R/database.R

```diff
@@ -1715,98 +1715,154 @@ get_sample_cn_segments = function(this_sample_id,
-  table_name_unmatched = config::get("results_tables")$copy_number_unmatched
+  
+  #check seq type and return a message if anything besides "genome" is called. To be updated once capture samples have been processed through CNV protocols.
+  if(this_seq_type!="genome"){
+    stop("Currently, only genome samples are available for this function. Please select a valid seq type (i.e genome). Compatibility for capture samples will be added soon...")
+  }
+  
+  #get wildcards from this_seq_type (lazy)
+  seq_type = this_seq_type
+  
+  #perform wrangling on the region to have it in the correct format. 
   if(!missing(region)){
     region = gsub(",", "", region)
-    #format is chr6:37060224-37151701
     split_chunks = unlist(strsplit(region, ":"))
     chromosome = split_chunks[1]
     startend = unlist(strsplit(split_chunks[2], "-"))
     qstart = startend[1]
     qend = startend[2]
   }
-  if(grepl("chr", chromosome)){
+  
+  #deal with chr prefixes for region, based on selected genome projection.
+  if(projection == "grch37"){
+    if(grepl("chr", chromosome)){
+      chromosome = gsub("chr", "", chromosome)
+    }
   }else{
-    chromosome = paste0("chr", chromosome)
+    if(!grepl("chr", chromosome)){
+      chromosome = paste0("chr", chromosome)
+    }
   }
-
-  #chr prefix the query chromosome to match how it's stored in the table.
-  #This isn't yet standardized in the db so it's just a workaround "for now".
+  
+  #sanity check type of qstart en qend
+  if(!is.numeric(qstart)){
+    qstart = as.numeric(qstart)
+  }
+  
+  if(!is.numeric(qend)){
+    qend = as.numeric(qend)
+  }
+  
   if(from_flatfile){
-    base_dir = config::get(config="default")$project_base
-
-    unmatched_path = config::get()$results_directories$controlfreec
-
-    #separated by which genome the sample was aligned to
-    unmatched_hg38_path = paste0(unmatched_path, "from--genome--hg38/")
-    unmatched_grch37_path = paste0(unmatched_path, "from--genome--grch37/")
+    cnv_flatfile_template = config::get("results_flatfiles")$cnv_combined$icgc_dart
+    cnv_path =  glue::glue(cnv_flatfile_template)
+    full_cnv_path =  paste0(config::get("project_base", config = "default"), cnv_path)
+    
+    #check permissions to ICGC data.
+    permissions = file.access(full_cnv_path, 4)
+    if(permissions == -1){
+      message("restricting to non-ICGC data")
+      cnv_flatfile_template = config::get("results_flatfiles")$cnv_combined$gambl
+      cnv_path =  glue::glue(cnv_flatfile_template)
+      full_cnv_path =  paste0(config::get("project_base"), cnv_path)
+    }
+    
+    #check for missingness.
+    if(!file.exists(full_cnv_path)){
+      print(paste("missing: ", full_cnv_path))
+      message("Cannot find file locally. If working remotely, perhaps you forgot to load your config (see below) or sync your files?")
+      message('Sys.setenv(R_CONFIG_ACTIVE = "remote")')
+    }
+        
+    all_segs = all_segs = suppressMessages(read_tsv(full_cnv_path)) %>%
```

Comment:
This here has double assignment, maybe result of the merge? Needs to be resolved - not sure how it will behave

Reply:
Oh, yes. Must've been a merge result. I'll go over all of these. Good catch!